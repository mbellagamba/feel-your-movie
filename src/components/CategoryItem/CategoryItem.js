import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { stringToColor } from '../../utils';
import { RADIUS_CARD } from '../../resources/dimensions';

const Container = styled.div`
  width: 10rem;
  height: 3rem;
  background-color: #${(props) => props.color}44;
  margin: 0.5em;
  display: flex;
  text-align: center;
  flex-direction: column;
  justify-content: center;
  border-radius: ${RADIUS_CARD};
`;

const Text = styled.span`
  font-weight: 500;
`;

const CategoryItem = ({ name }) => (
  <Container color={stringToColor(name)}>
    <Text>{name}</Text>
  </Container>
);

CategoryItem.propTypes = {
  name: PropTypes.string.isRequired,
};

export default CategoryItem;
