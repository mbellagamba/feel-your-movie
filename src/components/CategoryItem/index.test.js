import React from 'react';
import { render } from '@testing-library/react';
import CategoryItem from './index';

test('should show the given name', () => {
  const name = 'test';
  const { getByText } = render(<CategoryItem index={0} name={name} />);
  expect(getByText(name)).toBeInTheDocument();
});

test('should match snapshot', () => {
  const { asFragment } = render(<CategoryItem index={0} name="test" />);
  expect(asFragment()).toMatchSnapshot();
});
