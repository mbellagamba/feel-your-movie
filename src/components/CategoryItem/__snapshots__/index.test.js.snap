// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  min-width: 12rem;
  height: 3rem;
  background-image: linear-gradient( to right, #1abc9c 0%, #15987e 51%, #107461 100% );
  margin: 0.5em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
}

.c0:hover {
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  -webkit-transition: all 0.4s;
  transition: all 0.4s;
}

.c1 {
  font-weight: 500;
}

<div
    class="c0"
    color="#1abc9c"
  >
    <h4
      class="c1"
    >
      test
    </h4>
  </div>
</DocumentFragment>
`;
